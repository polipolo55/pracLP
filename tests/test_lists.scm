(define (main)
  (display "Prova de creaci√≥ de llistes:")
  (newline)
  (display "Creant llista '(1 2 3): ")
  (display '(1 2 3))
  (newline)

  (display "Prova de car amb llista '(1 2 3): ")
  (display (car '(1 2 3)))
  (newline)

  (display "Prova de cdr amb llista '(1 2 3): ")
  (display (cdr '(1 2 3)))
  (newline)

  (display "Prova de cons: cons (1 2 3): ")
  (display (cons 1 '(2 3)))
  (newline)

  (display "Prova de null? amb '()': ")
  (display (null? '()))
  (newline)

  (display "Prova de null? amb '(1 2 3)': ")
  (display (null? '(1 2 3)))
  (newline)

  (display "Totes les proves de llistes han finalitzat!")
  (newline)
)