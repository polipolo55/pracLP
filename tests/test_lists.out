Prova de creaci√≥ de llistes:

Creant llista '(1 2 3): 
[1, 2, 3]

Prova de car amb llista '(1 2 3): 
1

Prova de cdr amb llista '(1 2 3): 
[2, 3]

Prova de cons: cons (1 2 3): 
[1, 2, 3]

Prova de null? amb '()': 
True

Prova de null? amb '(1 2 3)': 
False

Totes les proves de llistes han finalitzat!

